### PR TITLE
Adding dotnet isolated image to the core-tools nightly build

### LINKED ISSUE #394: Core-Tools Nightly Build fails with DotNet image 5.
https://azure-functions.visualstudio.com/azure-functions-docker/_build/results?buildId=7990&view=logs&j=6c314701-959d-5fd0-416e-8174fd0cc497&t=4d425a9b-3a28-58a6-acb9-2d8644e0627c

```
failed to solve with frontend dockerfile.v0: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount730593061/dotnet-core-tools.Dockerfile: no such file or directory
```

Need investigation, however, it looks .NET 5 related updated is not done on this pipeline.

### PR DESCRIPTION
**Pull request by TsuyoshiUshio**
Fix : https://github.com/Azure/azure-functions-docker/issues/394

Since the dotnet-isolated imaged is added, the directory structure has not been changed. 
This Pipeline also add the feature to publish the dotnet 5 isolated core tools images. 

@ankitkumarr Could you have a look?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,10 +14,20 @@ Linux amd64 Tags
...
+
+
+`mcr.microsoft.com/azure-functions/dotnet-isolated`
+
+| Tags                                       | Dockerfile                                                              | OS Version |
+|--------------------------------------------|-------------------------------------------------------------------------|------------|
+| `3.0`                                      | [Dockerfile](host/3.0/buster/amd64/dotnet/dotnet-isolated/dotnet-isolated.Dockerfile)            | Debian 10  |
+| `3.0-dotnet-isolated5.0-slim`                                 | [Dockerfile](host/3.0/buster/amd64/dotnet/dotnet-isolated/dotnet-slim.Dockerfile)       | Debian 10  |
+| `3.0-appservice`, `3.0-dotnet-isolated5.0-appservice` | [Dockerfile](host/3.0/buster/amd64/dotnet/dotnet-isolated/dotnet-appservice.Dockerfile) | Debian 10  |
+| `3.0-dotnet-isolated5.0-core-tools` | [Dockerfile](host/3.0/buster/amd64/dotnet/dotnet-isolated/-isolated-core-tools.Dockerfile) | Debian 10  |
```
Comment (ankitkumarr) [inline README.md]: Thanks for adding this!! :) Ahmed had asked me, but it slipped my mind.

File: host/3.0/core-tools-publish.yml [hunk 2]
```diff
@@ -72,6 +72,48 @@ stages:
...
+        env:
+          pswd: $(dockerPassword)
+
+      - bash: |
+          set -e
+          IMAGE_NAME=azurefunctions.azurecr.io/azure-functions/3.0/dotnet-isolated:$(PrivateVersion)-dotnet-isolated5.0-core-tools
+    
+          docker build -t $IMAGE_NAME \
+                      -f host/3.0/buster/amd64/dotnet/dotnet-isolated/dotnet-isolated-core-tools.Dockerfile \
+                      host/3.0/buster/amd64/dotnet/
```
Comment (ankitkumarr) [inline host/3.0/core-tools-publish.yml]: would this change to `host/3.0/buster/amd64/dotnet/dotnet-isolated/`?

File: host/3.0/core-tools-publish.yml [hunk 3]
```diff
@@ -63,7 +63,7 @@ stages:
           IMAGE_NAME=azurefunctions.azurecr.io/azure-functions/3.0/dotnet:$(PrivateVersion)-dotnet3-core-tools
     
           docker build -t $IMAGE_NAME \
-                      -f host/3.0/buster/amd64/dotnet/dotnet-core-tools.Dockerfile \
+                      -f host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-core-tools.Dockerfile \
                       host/3.0/buster/amd64/dotnet/
```
Comment (ankitkumarr) [inline host/3.0/core-tools-publish.yml]: would this change to `host/3.0/buster/amd64/dotnet/dotnet-inproc/`?

File: host/3.0/core-tools-publish.yml [hunk 4]
```diff
@@ -343,38 +385,72 @@ stages:
     
       - bash: |
           set -e
-          docker pull $SOURCE_REGISTRY/dotnet:$(PrivateVersion)-dotnet3-core-tools
+          docker pull $SOURCE_REGISTRY/dotnet:$(PrivateVersion)-dotnet-isolated5.0-core-tools
```
Comment (ankitkumarr) [inline host/3.0/core-tools-publish.yml]: I think these would be "$SOURCE_REGISTRY/**dotnet-isolated**:$(PrivateVersion)-dotnet-isolated5.0-core-tools". Same throughout this file.

File: README.md (hunk 1 above)
Comment (TsuyoshiUshio) [reply README.md]: You are welcome!

File: host/3.0/core-tools-publish.yml (hunk 3 above)
Comment (TsuyoshiUshio) [reply host/3.0/core-tools-publish.yml]: Done. Very good catch

File: host/3.0/core-tools-publish.yml (hunk 4 above)
Comment (TsuyoshiUshio) [reply host/3.0/core-tools-publish.yml]: Very good catch, I fixed wrong place.  Could you review this place again? @ankitkumarr

File: host/3.0/core-tools-publish.yml (hunk 2 above)
Comment (TsuyoshiUshio) [reply host/3.0/core-tools-publish.yml]: Very good catch. Done

**Review by pragnagopa — APPROVED** [review]
LGTM. Please wait for approval from @ankitkumarr

File: host/3.0/core-tools-publish.yml [hunk 5]
```diff
@@ -374,7 +416,41 @@ stages:
...
         continueOnError: false
         condition: ne(variables['SkipProduction'], 'true')
-        
+
+      - bash: |
+          set -e
+          docker pull $SOURCE_REGISTRY/dotnet-isolated:$(PrivateVersion)-dotnet-isolated5.0-core-tools
+
+          docker tag $SOURCE_REGISTRY/dotnet-isolated:$(PrivateVersion)-dotnet-isolated5.0-core-tools $TARGET_REGISTRY/dotnet:$(TargetVersion)-dotnet-isolated5.0-core-tools
+          docker tag $SOURCE_REGISTRY/dotnet-isolated:$(PrivateVersion)-dotnet-isolated5.0-core-tools $TARGET_REGISTRY/dotnet:$(MajorVersion)-dotnet-isolated5.0-core-tools
```
Comment (ankitkumarr) [inline host/3.0/core-tools-publish.yml]: Should this and above be "docker tag $SOURCE_REGISTRY/dotnet-isolated:$(PrivateVersion)-dotnet-isolated5.0-core-tools $TARGET_REGISTRY/**dotnet-isolated**:$(MajorVersion)-dotnet-isolated5.0-core-tools"?

**Review by ankitkumarr — APPROVED** [review]
Just one more comment. Looks good otherwise.

File: host/3.0/core-tools-publish.yml (hunk 5 above)
Comment (TsuyoshiUshio) [reply host/3.0/core-tools-publish.yml]: Thanks! good catch!

**Pull request merged**